pkg/segment/query/processor/streamstatscommand.go

```diff
@@ -18,26 +18,942 @@
+		}
+		if shouldResetBefore {
+			resetAccumulatedStreamStats(p.options)
+			currIndex = 0
+			numPrevSegmentProcessedRecords = 0
+		}
+
+		var timeInMilli uint64
+		if p.options.TimeWindow != nil {
+			tsVal, exists := record["timestamp"]
+			if !exists {
+				return nil, fmt.Errorf("timestamp required for time window but not found")
+			}
+			timeInMilli, err = dtypeutils.ConvertToUInt(tsVal, 64)
+			if err != nil {
+				return nil, fmt.Errorf("invalid timestamp value: %v", tsVal)
+			}
+		}
+
+		// Process each measure operation
+		for measIdx, measureAgg := range measureAggs {
+			if _, ok := p.options.RunningStreamStats[measIdx]; !ok {
+				p.options.RunningStreamStats[measIdx] = make(map[string]*structs.RunningStreamStatsResults)
+			}
+			if _, ok := p.options.RunningStreamStats[measIdx][bucketKey]; !ok {
+				p.options.RunningStreamStats[measIdx][bucketKey] = InitRunningStreamStatsResults(measureAgg.MeasureFunc)
+			}
+
+			colValue := CreateCValueFromColValue(record[measureAgg.MeasureCol])
+			fieldToValue := make(map[string]segutils.CValueEnclosure)
+			if measureAgg.ValueColRequest != nil {
+				for _, field := range measureAgg.ValueColRequest.GetFields() {
+					if values, ok := allColumns[field]; ok && i < len(values) {
+						fieldToValue[field] = values[i]
+					}
+				}
+			}
+
+			finalColValue, include := CreateCValueFromValueExpression(measureAgg, fieldToValue, colValue)
+
+			var result segutils.CValueEnclosure
+			var exists bool
+
+			if p.options.Window == 0 && p.options.TimeWindow == nil {
+				result, exists, err = PerformNoWindowStreamStatsOnSingleFunc(
+					p.options,
+					p.options.RunningStreamStats[measIdx][bucketKey],
+					measureAgg,
+					finalColValue,
+					include,
+				)
+			} else {
+				result, exists, err = PerformWindowStreamStatsOnSingleFunc(
+					int(numPrevSegmentProcessedRecords)+currIndex,
+					p.options,
+					p.options.RunningStreamStats[measIdx][bucketKey],
+					int(p.options.Window),
+					measureAgg,
+					finalColValue,
+					timeInMilli,
+					true, // timeSortAsc
+					include,
+				)
+			}
+
+			if err != nil {
+				return nil, fmt.Errorf("streamstats: error processing measure %v: %v", measureAgg.String(), err)
+			}
+
+			resultCol := measureAgg.String()
+			if exists {
+				knownValues[resultCol][i] = result
+			} else {
+				if measureAgg.MeasureFunc == utils.Count || measureAgg.MeasureFunc == utils.Cardinality {
+					knownValues[resultCol][i] = segutils.CValueEnclosure{
+						Dtype: utils.SS_DT_FLOAT,
+						CVal:  float64(0),
+					}
+				} else {
+					knownValues[resultCol][i] = segutils.CValueEnclosure{
+						Dtype: utils.SS_DT_STRING,
+						CVal:  "",
+					}
+				}
+			}
+		}
+
+		shouldResetAfter, err := evaluateResetCondition(p.options.ResetAfter, record)
+		if err != nil {
+			return nil, err
+		}
+		if shouldResetAfter {
+			resetAccumulatedStreamStats(p.options)
+			currIndex = 0
+			numPrevSegmentProcessedRecords = 0
+		}
+
+		p.options.NumProcessedRecords++
```

Comment:
These parameters should be updated before reset. If reset happens we shouldn't increment them. Refer the older implementation.